### ISSUE TITLE
How do you use the relay node ID for filtering?

### ISSUE THREAD

**Issue by gojanpaolo** [ChilliCream/hotchocolate#2971]
Assuming I have the following types where the `Id`s implements relay node.

```c#
class Track
{
    public int Id { get; set; }
}
class Session
{
    public int TrackId { get; set; }
}
```

Is there a way to use the `Track.Id` from 

```graphql
query {
  tracks {
    id
  }
}
```

to filter `Session` by `TrackId` like so:

```
query {
  sessions (where: { trackId: { eq: "VHJhY2sKaTI=" } }) {
    ...
  }
}
```

The query above currently throws an error because it expects an `int` instead of an ID/string that is returned in `query { tracks { id } } `

## Use case

In our app, users will have the ability to filter `sessions` by selecting other related entities (e.g. `track`, `location`, etc). As mentioned, there are a number of related entities and we'd like to use them together so `GetSessionByTrackId`, `GetSessionByLocaionId` will not work. Without implementing relay node, we could directly use the related entities' `id`s to filter the `sessions`.

In other words, we're looking into a similar graphql query that will behave like this c# code

```c#
sessions
.Where(s => s.TrackId == trackId)
.Where(s => s.LocationId == locationId)
// ... more filter for other related entities if the user have selected it
```

Thank you!

**Comment by TDDinAction** [comment]
@PascalSenn https://github.com/ChilliCream/hotchocolate/pull/2999

It seems merging of this support was blocked. It turns out this support will be included in the 12.0.0 version?

**Comment by chungonion** [comment]
A bit curious that whether there are updates on this issue? Thank you!

**Comment by PascalSenn** [comment]
This is now possible, but you have to define the filter type:

```csharp
public class FooFilterInputType:  FilterInputType<Foo>
{
    protected override void Configure(IFilterInputTypeDescriptor<Foo> descriptor) 
    {
           descriptor.Field(y => y.Bar).Type<IdOperationFilterInputType>())
    }
}
```

**Comment by VassliyBychko** [comment]
When using `IdOperationFilterInputType` with Guid id, it tries to deserialize guid part of relay id as a string, so it gets something like: "�tǭ(�\u0014L���X��!" and then fails with message: "Unable to convert type from `Object` to `Guid`". Would you mind helping me with this issue? Thank you!

**Comment by tomachristian** [comment]
We are having the same problem as @VassliyBychko
